File: StringCompression.js
var element=(word)=>{
    let comp=new Array();
    let index=1;
    let result='';
    comp.push(word.charAt(0));
    while(index<word.length)
    {
        if(comp.length==9 || comp.some(element=>element!==word[index]))
        {
            result+=comp.length.toString()+comp[0];
            comp=new Array();

        }
        comp.push(word[index]);
        index+=1
    

    }
    if(comp.length!==0){
        result+=comp.length.toString()+comp[0]
    }

    return result;
}

function main(){
    var result=element("aaabbbcccccccccddddd");
    console.log(result);
}
main()

File: StringCompression.java
import java.util.*;
class StringCompression{
    public static void main(String[] args) {
        String input="aaaaaaaaaaaaaabb";
        System.out.println(new StringCompression().compressedString(input));
    }

    public String compressedString(String word)
    {
        List<String> list=new LinkedList();
        int index=1;
        String result="";
        list.add(word.charAt(0)+"");
        while(index<word.length())
        {
            if(list.size()==9 || ! list.contains(word.charAt(index)+"")){
                result+=list.size()+""+list.get(0);
                list.clear();
            }
            list.add(word.charAt(index)+"");
            index+=1;
        }
        if( list.size()!=0){
            result+=list.size()+""+list.get(0);
        }
        return result;
    }
}